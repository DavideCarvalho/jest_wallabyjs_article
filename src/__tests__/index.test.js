const api = require("../index");
const axios = require("axios");

const mockedAxios = {
  data: {
    userId: 1,
    id: 1,
    title:
      "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    body:
      "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
  }
};

describe("index.js", () => {
  it("adds 1 + 2 to equal 3", () => {
    expect(api.somaDoisNumeros(1, 2)).toMatchSnapshot();
  });

  it("expect to mock axios request", async () => {
    const spy = jest.spyOn(axios, "get");
    spy.mockReturnValue(mockedAxios);
    const apiResponse = await api.pegaPost(1);
    expect(apiResponse).toMatchSnapshot();
  });
});
