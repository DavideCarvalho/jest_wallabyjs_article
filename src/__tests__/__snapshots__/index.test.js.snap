// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.js adds 1 + 2 to equal 3 1`] = `3`;

exports[`index.js expect to mock axios request 1`] = `
"otcetihcra teineve mer tnus metua tse murer murtson
matot sauq tu tu eaitselom tiredneherper
muc te atidepxe rutnuuqesnoc eadnasucer tipicsus
tipicsus te aiuq"
`;
